### PR TITLE
feature: Collect and return ranges of paragraphs #777

### PR DESCRIPTION
**Pull request by Nurkambay**
To build the CCF graph we need to have ranges of paragraphs and sections to check the usages belongs to the paragraph or to the section.

- 2 fields to the result added
- unit tests added
- use case tests added

### CODE REVIEW

File: server/src/main/java/com/broadcom/lsp/cobol/core/model/RangeModel.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+package com.broadcom.lsp.cobol.core.model;
+
+import lombok.Value;
+import org.eclipse.lsp4j.Range;
+
+/**
+ * This model represents a range of a statement (i.e. section or paragraph)
+ */
+@Value
+public class RangeModel {
```
Comment (grianbrcom) [inline server/src/main/java/com/broadcom/lsp/cobol/core/model/RangeModel.java:24]: The class content confuses me. Looks like it is a range with start and end. But it consists of ranges.
What do you expect as a result of `rangeModel.getStart().getEnd()` or `rangeModel.getEnd().getStart()`?

**Review by grianbrcom — CHANGES_REQUESTED** [review]
Please, fix URI related comments.

File: server/src/main/java/com/broadcom/lsp/cobol/core/visitor/CobolVisitor.java [hunk 2]
```diff
@@ -239,10 +240,37 @@ public Void visitProcedureSection(ProcedureSectionContext ctx) {
...
+    }
+  }
+
+  private RangeModel getRange(ParserRuleContext ctx) {
+    String name = ctx.getStart().getText().toUpperCase();
+    Optional<Locality> start = getLocality(ctx.getStart());
+    Optional<Locality> end = getLocality(ctx.getStop());
+
+    if (start.isPresent() && end.isPresent()) {
+      return new RangeModel(name, start.get().getRange(), end.get().getRange());
```
[the remark this answers is not in the document]
Comment (Nurkambay) [reply server/src/main/java/com/broadcom/lsp/cobol/core/visitor/CobolVisitor.java:269]: Got it, thanks

**Review by grianbrcom — APPROVED** [review]
:leafy_green:

**Pull request merged**